cms/toolbar/toolbar.py

```diff
@@ -97,7 +98,7 @@ def init_toolbar(self, request):
         self.request = request
         self.is_staff = self.request.user.is_staff
         self.edit_mode = self.is_staff and self.request.session.get('cms_edit', False)
-        self.build_mode = self.is_staff and self.request.session.get('cms_build', False)
+        self.build_mode = self.is_staff and request.GET.get(get_cms_setting('CMS_TOOLBAR_URL__BUILD')) != None
```

Comment:
My feeling is that we're not doing this right. Since changing between structure and preview mode does not require a page load, I don't think it should have a GET parameter to do, but rather a # parameter.